parachain/pallets/verifier-lightclient/src/lib.rs

```diff
@@ -124,7 +129,42 @@ impl<T: Trait> Module<T> {
 			Error::<T>::DuplicateHeader,
 		);
 
-		// TODO check PoW
+		if !VerifyPoW::get() {
+			return Ok(());
+		}
+
+		// Adapted from https://github.com/near/rainbow-bridge/blob/3fcdfbc6c0011f0e1507956a81c820616fb963b4/contracts/near/eth-client/src/lib.rs#L363
+		// See YellowPaper formula (50) in section 4.3.4
+		let parent = Headers::<T>::get(header.parent_hash)
+			.ok_or(Error::<T>::MissingParentHeader)?
+			.header;
+		ensure!(
+			header.gas_used <= header.gas_limit
+			&& header.gas_limit < parent.gas_limit * 1025 / 1024
+			&& header.gas_limit > parent.gas_limit * 1023 / 1024
+			&& header.gas_limit >= 5000.into()
+			&& header.timestamp > parent.timestamp
+			&& header.number == parent.number + 1
+			&& header.extra_data.len() <= 32,
+			Error::<T>::InvalidHeader,
+		);
+
+		// Simplified difficulty check to conform adjusting difficulty bomb
+		let header_mix_hash = header.mix_hash().ok_or(Error::<T>::InvalidHeader)?;
+		let header_nonce = header.nonce().ok_or(Error::<T>::InvalidHeader)?;
+		let (mix_hash, result) = EthashProver::new().hashimoto_merkle(
+			header.compute_partial_hash(),
+			header_nonce,
+			header.number,
+			proof,
+		).map_err(|e| Error::<T>::InvalidHeader)?;
```

Comment:
Linter is going to complain about `e` not being used.

```suggestion
		).map_err(|_| Error::<T>::InvalidHeader)?;
```